sdk/tests/position.test.ts

```diff
@@ -0,0 +1,255 @@
+let token0 = await deployPSP22(api, account, 1000000000n, 'Coin', 'COIN', 0n, Network.Local)
+let token1 = await deployPSP22(api, account, 1000000000n, 'Coin', 'COIN', 0n, Network.Local)
+
+const lowerTickIndex = -20n
+const upperTickIndex = 10n
+const feeTier = newFeeTier({ v: 6000000000n }, 10n)
+
+let poolKey = newPoolKey(
+  token0.contract.address.toString(),
+  token1.contract.address.toString(),
+  feeTier
+)
+
+describe('position', async () => {
+  beforeEach(async () => {
+    invariant = await deployInvariant(api, account, { v: 10000000000n }, Network.Local)
+    token0 = await deployPSP22(api, account, 1000000000n, 'Coin', 'COIN', 0n, Network.Local)
+    token1 = await deployPSP22(api, account, 1000000000n, 'Coin', 'COIN', 0n, Network.Local)
+
+    poolKey = newPoolKey(
+      token0.contract.address.toString(),
+      token1.contract.address.toString(),
+      feeTier
+    )
+
+    await invariant.addFeeTier(account, feeTier)
+
+    await invariant.createPool(
+      account,
+      token0.contract.address.toString(),
+      token1.contract.address.toString(),
+      feeTier,
+      { v: 1000000000000000000000000n },
+      0n
+    )
+
+    await token0.approve(account, invariant.contract.address.toString(), 10000000000n)
+    await token1.approve(account, invariant.contract.address.toString(), 10000000000n)
+
+    const pool = await invariant.getPool(
+      account,
+      token0.contract.address.toString(),
+      token1.contract.address.toString(),
+      feeTier
+    )
+
+    const result = await invariant.createPosition(
+      account,
+      poolKey,
+      lowerTickIndex,
+      upperTickIndex,
+      { v: 1000000000000n },
+      pool.sqrtPrice,
+      pool.sqrtPrice
+    )
+
+    assert.deepEqual(result.events[0].address, account.address.toString())
+    assert.deepEqual(result.events[0].currentSqrtPrice, {
+      v: 1000000000000000000000000n
+    })
+    assert.deepEqual(result.events[0].liquidity, { v: 1000000000000n })
+    assert.deepEqual(result.events[0].lowerTick, -20n)
+    assert.deepEqual(result.events[0].pool.tokenX, poolKey.tokenX)
+    assert.deepEqual(result.events[0].pool.tokenY, poolKey.tokenY)
+    assert.deepEqual(result.events[0].upperTick, 10n)
+  })
+
+  it('create position', async () => {
+    const position = await invariant.getPosition(account, account.address, 0n)
+    const expectedPosition: Position = {
+      poolKey: poolKey,
+      liquidity: { v: 1000000000000n },
+      lowerTickIndex: lowerTickIndex,
+      upperTickIndex: upperTickIndex,
+      feeGrowthInsideX: { v: 0n },
+      feeGrowthInsideY: { v: 0n },
+      lastBlockNumber: 0n,
+      tokensOwedX: 0n,
+      tokensOwedY: 0n
+    }
+    await positionEquals(position, expectedPosition)
+  })
+
+  it('remove position', async () => {
+    {
+      const result = await invariant.removePosition(account, 0n)
+
+      assert.deepEqual(result.events[0].address, account.address.toString())
```

Comment:
Adding the util function for comparing events would be helpful here. it make tests code more readable and , we will likely need to compare events when testing stats or event subscription